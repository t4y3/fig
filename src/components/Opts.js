import { h } from 'hyperapp';

/**
 * format opts value
 */
const format = (e, data, key) => {
  const djt = new DomJsonTree(
    {
      [key]: data,
    },
    e,
  );
  djt.render();
};

const Opts = ({ opts = {} }) => {
  const isEmpty = !Object.keys(opts).length;
  if (isEmpty) {
    return null;
  }
  return (
    <div className="fig-opts">
      <div className="siimple-h6">Opts</div>
      <div className="siimple-table siimple-table--border">
        <div className="siimple-table-header">
          <div className="siimple-table-row">
            <div className="siimple-table-cell siimple--width-25">Opts Name</div>
            <div className="siimple-table-cell siimple--width-75">Value</div>
          </div>
        </div>
        <div className="siimple-table-body">
          {Object.keys(opts).map(key => (
            <div className="siimple-table-row">
              <div className="siimple-table-cell siimple--width-25">{key}</div>
              <div
                className="siimple-table-cell siimple--width-75"
                oncreate={(e) => {
                  format(e, opts[key], key);
                }}
              />
            </div>
          ))}
        </div>
      </div>
    </div>
  );
};

export default Opts;
