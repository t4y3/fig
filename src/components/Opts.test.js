import { h } from 'hyperapp';
import { renderToString } from 'hyperapp-render';
import Opts from './Opts';

test('Opts.js to match snapshot', () => {
  const opts = {
    a: 1,
    b: 'test',
    c: true,
    d: {
      a: 1,
      b: 'test'
    },
    e: [1, 2, 3]
  };
  const view = renderToString(<Opts opts={opts} />);
  expect(view).toMatchSnapshot();
});
