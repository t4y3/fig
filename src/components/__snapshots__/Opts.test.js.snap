// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Opts.js to match snapshot 1`] = `"<div className=\\"fig-opts\\"><div className=\\"siimple-h6\\">Opts</div><div className=\\"siimple-table siimple-table--border\\"><div className=\\"siimple-table-header\\"><div className=\\"siimple-table-row\\"><div className=\\"siimple-table-cell siimple--width-25\\">Opts Name</div><div className=\\"siimple-table-cell siimple--width-75\\">Value</div></div></div><div className=\\"siimple-table-body\\"><div className=\\"siimple-table-row\\"><div className=\\"siimple-table-cell siimple--width-25\\">a</div><div className=\\"siimple-table-cell siimple--width-75\\"></div></div><div className=\\"siimple-table-row\\"><div className=\\"siimple-table-cell siimple--width-25\\">b</div><div className=\\"siimple-table-cell siimple--width-75\\"></div></div><div className=\\"siimple-table-row\\"><div className=\\"siimple-table-cell siimple--width-25\\">c</div><div className=\\"siimple-table-cell siimple--width-75\\"></div></div><div className=\\"siimple-table-row\\"><div className=\\"siimple-table-cell siimple--width-25\\">d</div><div className=\\"siimple-table-cell siimple--width-75\\"></div></div><div className=\\"siimple-table-row\\"><div className=\\"siimple-table-cell siimple--width-25\\">e</div><div className=\\"siimple-table-cell siimple--width-75\\"></div></div></div></div></div>"`;
